sklearn/ensemble/forest.py

```diff
@@ -374,18 +374,17 @@ def feature_importances_(self):
         return sum(all_importances) / len(self.estimators_)
 
 
-# these next two are utility functions for joblib's Parallel. Ideally
-# they would be defined locally in ForestClassifier or ForestRegressor, but
-# joblib complains that it cannot Pickle them when placed there.
-
-def _run_estimator(f, X, out):
-    out += f(X, check_input=False)
-
-
-def _run_estimator2(f, X, out):
-    all_proba = f(X, check_input=False)
-    for i in range(len(out)):
-        out[i] += all_proba[i]
+# This is a utility function for joblib's Parallel. It can't go locally in
+# ForestClassifier or ForestRegressor, because joblib complains that it cannot
+# pickle it when placed there.
+
+def accumulate_prediction(predict, X, out):
+    prediction = predict(X, check_input=False)
+    if len(out) == 1:
```

Comment:
In this revised function, `out` is always a list. It contains only one element in the single output case. The function could 

1. accept either a list of arrays or a single array, or 
2. only accept a list of arrays.

I figured (2) was simpler, but I can change to (1) if desired.